src/projectManager.ts

```diff
@@ -2,72 +2,74 @@
  * (c) Copyright IBM Corp. 2023
  */
 
-import { QuickPickItem, Uri, window, workspace, WorkspaceFolder } from "vscode";
+import { ExtensionContext, QuickPickItem, StatusBarAlignment, StatusBarItem, Uri, window, workspace, WorkspaceFolder } from "vscode";
 import { IProject } from "./iproject";
 import { ProjectExplorerTreeItem } from "./views/projectExplorer/projectExplorerTreeItem";
 import Project from "./views/projectExplorer/project";
 
 export class ProjectManager {
     private static loaded: { [index: number]: IProject } = {};
+    private static activeProject: IProject | undefined;
+    private static activeProjectStatusBarItem: StatusBarItem;
 
     public static load(workspaceFolder: WorkspaceFolder) {
         if (!this.loaded[workspaceFolder.index]) {
-            this.loaded[workspaceFolder.index] = new IProject(workspaceFolder);
+            const iProject = new IProject(workspaceFolder);
+            this.loaded[workspaceFolder.index] = iProject;
+
+            if (!this.activeProject) {
```

Comment:
This method is to load all projects into the workspace either at the start or when new workspace folders get added. We only set `this.activeProject` in this method if one is not already defined. Thus, we do not need to do any comparison here.

Reply:
@SanjulaGanepola But based on this comparison you are setting the active project to the current workspaceFolder
So wouldnt the active Project end up with the last workspaceFolder set, rather than the one that correspondes with the activeProject?